Update index page (#5869)
nextflow-io/nextflow: docs/index.md

[![install with bioconda](https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?colorB=58bd9f&style=popout)](http://bioconda.github.io/recipes/nextflow/README.html)
[![Nextflow license](https://img.shields.io/github/license/nextflow-io/nextflow.svg?colorB=58bd9f&style=popout)](https://github.com/nextflow-io/nextflow/blob/master/COPYING)

> Nextflow is a workflow system for creating scalable, portable, and reproducible workflows. It is based on the dataflow programming model that simplifies writing parallel and distributed pipelines and allows you to focus on data flow and computation. Nextflow can deploy workflows on a variety of execution platforms, including your local machine, HPC schedulers, and cloud. Additionally, Nextflow supports a range of software environment and package managers and allows workflows to be executed in reproducible and isolated environments.

Reviewer: ```suggestion
Nextflow is a workflow system for creating scalable, portable, and reproducible workflows. It uses a dataflow programming model that simplifies writing parallel and distributed pipelines by allowing you to focus on data flow and computation. Nextflow can deploy workflows on a variety of execution platforms, including your local machine, HPC schedulers, and cloud. Additionally, Nextflow supports a range of compute environments and package managers, allowing workflows to be executed in reproducible and isolated environments.
```

Reviewer 2: it would be nice to mention "software container runtimes" (or engine). It's a peculiar capability
